File: server/src/main/java/se/kth/id2203/core/OriginatedData.scala
package se.kth.id2203.core

import se.sics.kompics.KompicsEvent
import se.sics.kompics.network.Address

/**
  * Created by sindrikaldal on 17/02/17.
  */
case class OriginatedData(src: Address, payload: KompicsEvent) extends KompicsEvent;

File: server/src/main/java/se/kth/id2203/broadcasting/OriginatedData.java
package se.kth.id2203.broadcasting;

import se.kth.id2203.networking.NetAddress;
import se.sics.kompics.KompicsEvent;

import java.io.Serializable;

/**
 * Created by sindrikaldal on 23/02/17.
 */
public class OriginatedData implements KompicsEvent, Serializable {
    public final NetAddress src;
    public final KompicsEvent payload;

    public OriginatedData(NetAddress src, KompicsEvent payload) {
        this.src = src;
        this.payload = payload;
    }
}
